src/common/gpx.c

```diff
@@ -151,6 +155,55 @@ void dt_gpx_destroy(struct dt_gpx_t *gpx)
   g_free(gpx);
 }
 
+/* --------------------------------------------------------------------------
+ * Geodesic interpolation functions
+ * ------------------------------------------------------------------------*/
+
+ static void dt_gpx_geodesic_distance(double lat1, double lon1,
+                      double lat2, double lon2,
+                      double *d, double *delta
+                    )
+{
+  const double lat_rad_1 = lat1 * PI / 180;
+  const double lat_rad_2 = lat2 * PI / 180;
+  const double lon_rad_1 = lon1 * PI / 180;
+  const double lon_rad_2 = lon2 * PI / 180;
+  const double delta_lat_rad = lat_rad_2 - lat_rad_1;
+  const double delta_lon_rad = lon_rad_2 - lon_rad_1;
+
+  const double a = sin(delta_lat_rad / 2) * sin(delta_lat_rad / 2) +
```

Comment:
please do something like 
```C
const double sin_delta_lat_rad = sin(delta_lat_rad / 2);
const double a = sin_delta_lat_rad * sin_delta_lat_rad + ...
```
Transcendental functions like cos and sin are expensive to compute, and here you recompute them several times. It's more energy-efficient to store them the first time in a variable and simply reuse that variable later.